Makefile

```diff
@@ -0,0 +1,130 @@
+# SPDX-License-Identifier: CC0-1.0
+#
+# SPDX-FileContributor: Antonio Niño Díaz, 2023
+
+# Source code paths
+# -----------------
+
+SOURCEDIRS	:= src
+INCLUDEDIRS	:= src
+
+# Libraries
+# ---------
+
+LIBS		:= -lm
+LIBDIRS		:=
+
+# Build artifacts
+# ---------------
+
+NAME		:= ptexconv
+BUILDDIR	:= build
+ELF		:= $(NAME)
+
+# Tools
+# -----
+
+STRIP		:= -s
+BINMODE		:= 755
+
+CC		:= gcc
+CXX		:= g++
+CP		:= cp
+MKDIR		:= mkdir
+RM		:= rm -rf
+MAKE		:= make
+INSTALL		:= install
+
+# Verbose flag
+# ------------
+
+ifeq ($(VERBOSE),1)
+V		:=
+else
+V		:= @
+endif
+
+# Source files
+# ------------
+
+SOURCES_C	:= $(shell find -L $(SOURCEDIRS) -name "*.c")
+SOURCES_CPP	:= $(shell find -L $(SOURCEDIRS) -name "*.cpp")
+
+# Compiler and linker flags
+# -------------------------
+
+WARNFLAGS_C	:= -Wall -Wextra -Wpedantic \
+		-Wno-maybe-uninitialized -Wno-alloc-size-larger-than -Wno-pointer-sign \
+		-Wno-unused-variable -Wno-unused-result -Wno-unused-parameter \
+		-Wno-unused-but-set-variable
+
+WARNFLAGS_CXX	:= -Wall -Wextra
+
+ifeq ($(SOURCES_CPP),)
+    LD	:= $(CC)
+else
+    LD	:= $(CXX)
+endif
+
+INCLUDEFLAGS	:= $(foreach path,$(INCLUDEDIRS),-I$(path)) \
+		   $(foreach path,$(LIBDIRS),-I$(path)/include)
+
+LIBDIRSFLAGS	:= $(foreach path,$(LIBDIRS),-L$(path)/lib)
+
+CFLAGS		+= -std=gnu11 $(WARNFLAGS_C) $(DEFINES) $(INCLUDEFLAGS) -O3
+
+CXXFLAGS	+= -std=gnu++14 $(WARNFLAGS_CXX) $(DEFINES) $(INCLUDEFLAGS) -O3
+
+LDFLAGS		:= $(LIBDIRSFLAGS) $(LIBS)
+
+# Intermediate build files
+# ------------------------
+
+OBJS		:= $(addsuffix .o,$(addprefix $(BUILDDIR)/,$(SOURCES_C))) \
+		   $(addsuffix .o,$(addprefix $(BUILDDIR)/,$(SOURCES_CPP)))
+
+DEPS		:= $(OBJS:.o=.d)
+
+# Targets
+# -------
+
+.PHONY: all clean install
+
+all: $(ELF)
+
+$(ELF): $(OBJS)
+	@echo "  LD      $@"
+	$(V)$(LD) -o $@ $(OBJS) $(LDFLAGS)
+
+clean:
+	@echo "  CLEAN  "
+	$(V)$(RM) $(ELF) $(BUILDDIR)
+
+INSTALLDIR	?= /opt/blocksds/external/ptexconv
```

Comment:
I'm not sure about whether to consider this tool a "core" tool or "external" tool. I don't really control the repository, so I'm a bit worried about adding it as a submodule of BlocksDS. I could leave it external (like the libraries NFlib and Nitro Engine) and simply add it to the readme so that users know that this tool exists and that they can install it.